### ISSUE TITLE
feature_request(links): Pelican templates

### ISSUE THREAD

**Issue by Kristinita** [remarkjs/remark-validate-links#33]
### 1. Summary

It would be nice, if remark-validate-links will support Pelican templates, which have the form `{variable}`.

Or at least ignored them.

### 2. Argumentation

I use in my Markdown articles these templates:

+ [**`{filename}`**](https://docs.getpelican.com/en/latest/content.html#linking-to-internal-content)
+ [**`{static}`**](https://docs.getpelican.com/en/latest/content.html#linking-to-static-files)
+ [**`{attach}`**](https://docs.getpelican.com/en/latest/content.html#attaching-static-files)

remark-validate-links show warnings. I need to disable remark-validate-links completely, that use templates.

### 3. Data

+ `Kira_remark-valdate-links__pelican_article_templates--non-valid_demo.md`

```md
[![KiraGoddess]({static}/KiraFolder/KiraGoddess.png)]({static}/KiraFolder/KiraGoddess.html)

```

### 4. Expected behavior

No warnings.

### 5. Actual behavior

```text
D:\Kristinita>remark Kira_remark-valdate-links__pelican_article_templates--non-valid_demo.md --use remark-validate-links --no-stdout
Kira_remark-valdate-links__pelican_article_templates--non-valid_demo.md
  1:1-1:92  warning  Link to unknown file: `%7Bstatic%7D/KiraFolder/KiraGoddess.html`  missing-file                     remark-validate-links
  1:2-1:53  warning  Link to unknown file: `%7Bstatic%7D/KiraFolder/KiraGoddess.png`   missing-file                     remark-validate-links

‼ 2 warnings
```

Thanks.

**Comment by ChristianMurphy** [comment]
Similar to https://github.com/remarkjs/remark-lint/issues/202
This would be a good parser extension for remark itself.

It may be possible to reuse https://github.com/djm/remark-shortcodes configured to use `{}` are the markers.
Then add a follow up plugin that can replace the shortcode with the variable/path as Pelican would resolve it.

**Comment by wooorm** [comment]
You could try something like this through a syntax extension indeed, for short codes. However, it’s impossible to totally support real template languages, as they could generate other markdown constructs.

For example, say we have a fictional template language where in `{foo}[link](url)` the `{foo}` part would expand to `!`, we’d now have an image instead of a link.

**Comment by wooorm** [comment]
Closing this as it’s not an issue that needs to be solved here!
